pkg/controller/controller.go

```diff
@@ -866,11 +867,25 @@ func convertAndFilterCatalogToNamespacedTypes(namespace string, in *osb.CatalogR
 	return serviceClasses, servicePlans, nil
 }
 
+func generateEscapedName(externalID string) string {
+	re := regexp.MustCompile("[^a-zA-Z0-9]")
+
+	escapedName := re.ReplaceAllStringFunc(externalID, func(in string) string {
+		return fmt.Sprintf("-%x-", in)
+	})
+
+	if len(escapedName) > 63 {
+		escapedName = escapedName[0:62]
```

Comment:
not thrilled but not sure what else to do - aside randomizing part of the string, but we can hold off on something that drastic until later I guess.

Reply:
re: hashing - yep. Jonathan and I were chatting about this the other day and I liked the idea of chopping the string when it's too long and replacing the tail of end it with a hash. At least then for cases where it's not too long they still get something readable.